.openshift-ci/get_latest_helm_chart_versions.py

```diff
@@ -84,6 +96,83 @@ def get_latest_helm_chart_version_for_specific_release(chart_name, release):
         remove_helm_repo()
 
 
+def get_supported_helm_chart_versions():
+    add_helm_repo()
+    try:
+        update_helm_repo()
+        return __get_supported_helm_chart_versions()
+    finally:
+        remove_helm_repo()
+
+
+def __get_supported_helm_chart_versions():
+    supported_central_versions = []
+    supported_sensor_versions = []
+
+    supported_releases = get_supported_releases()
+    for release in supported_releases:
+        if __does_chart_exist("stackrox-central-services", release):
+            supported_central_versions.append(__get_latest_helm_chart_version_for_specific_release(
+                "stackrox-central-services", release)
+            )
+        else:
+            logging.debug(f"Supported version \"{release.major}.{release.minor}\" has no corresponding helm chart for "
+                          f"stackrox-central-services.")
+        if __does_chart_exist("stackrox-secured-cluster-services", release):
+            supported_sensor_versions.append(__get_latest_helm_chart_version_for_specific_release(
+                "stackrox-secured-cluster-services", release)
+            )
+        else:
+            logging.debug(f"Supported version \"{release.major}.{release.minor}\" has no corresponding helm chart for "
+                          f"stackrox-secured-cluster-services.")
+    return supported_central_versions, supported_sensor_versions
+
+
+def get_supported_releases():
+    supported_releases = []
+    data = __get_data_from_product_lifecycles_api()
+    releases = []
+    try:
+        releases = data["data"][0]["versions"]
+    except Exception as exception:
+        logging.debug(f"Found no RHACS releases in PRODUCT_LIFECYCLES_API at "
+                      f"{PRODUCT_LIFECYCLES_API}\n{repr(exception)}")
+    for release in releases:
+        if release["type"] != "End of life":
+            supported_releases.append(parse_release(release["name"]))
+    return supported_releases
+
+
+def __get_data_from_product_lifecycles_api():
+    req = Request(
+        url=PRODUCT_LIFECYCLES_API,
+        headers={'User-Agent': 'Mozilla/5.0'}
+    )
+    try:
+        response = urlopen(req)
+        with response:
```

Comment:
```suggestion
        with urlopen(req) as response:
```

Reply:
I do have to split urlopen(req) out of the with if I want to put it into a try / except block. But I have restructured the code so it makes more sense.

Reply:
🤔 I tried with the code I suggested below I think it works inside the `try-catch`. I'm not sure if the `with` is used properly the way you have here. As far as I recall `with` is used with classes that have `__init__`, `__enter__`, `__exit__` defined or with functions that have a `yield` function. I think this scenario is the later so you need the `with urlopen(req) as request:` in order for the `with` to work properly.

Reply:
I've read up a bit about `with` in python, you're right. I'll put `urlopen(req)` back inside it and the try except block around it.